core/vm/evm.go

```diff
@@ -62,10 +61,21 @@ func (evm *EVM) precompile(addr common.Address) (PrecompiledContract, bool) {
 
 // run runs the given contract and takes care of running precompiles with a fallback to the byte code interpreter.
 func run(evm *EVM, contract *Contract, input []byte, readOnly bool) ([]byte, error) {
+	callback, err := selectInterpreter(evm, contract)
+	defer callback()
```

Comment:
probably err check must be before defer callback()

Reply:
has been changed